### PR TITLE
Switch view engine from jade to pug

### LINKED ISSUE #8743: s/jade/pug/g
replace jade package with newly renamed pug.
rename jade helper functions to pug helper functions
replace .jade extensions with .pug   🐶 

There may be other things that should be replaced that I'm missing

### PR DESCRIPTION
**Pull request by bugron**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [ ] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #8743
#### Description

So the hardest part was a breaking change. They've removed support for interpolation in attributes. I've refactored everything I could. Pledge and commit need to be QA'd though. If you find something like `superBlock.name` or `challenge.dashedName` in the text then ping me to fix it.

### CODE REVIEW

**Comment by bugron** [comment]
You're so kind, @camperbot ;)

File: package.json [hunk 1]
```diff
@@ -105,6 +104,7 @@
     "passport-oauth": "^1.0.0",
     "passport-twitter": "^1.0.3",
     "pmx": "~0.5.5",
+    "pug": "^2.0.0-alpha8",
```
Comment (BerkeleyTrue) [inline package.json]: Why is this alpha? Has the package changed that much?

Comment (bugron) [inline package.json]: Here are available npm versions: '0.0.0', '0.1.0', '2.0.0-alpha1', '2.0.0-alpha2', '2.0.0-alpha3', '2.0.0-alpha4', '2.0.0-alpha5', '2.0.0-alpha6', '2.0.0-alpha7', '2.0.0-alpha8'. There are indeed some breaking changes, see https://github.com/pugjs/pug/issues/2305.

Comment (BerkeleyTrue) [inline package.json]: I see... We'll should probably wait for stable to be released.

**Comment by BerkeleyTrue** [comment]
Looks like they hit Beta! We're almost there!

**Comment by hallaathrad** [comment]
closes #8743

**Comment by Lakston** [comment]
Does this still needs to be addressed ?

**Comment by BerkeleyTrue** [comment]
Going to close this. It doesn't seem like there pug is moving out of beta anytime soon.

Happy Coding

**Pull request closed without merging**